### PR TITLE
upgrade webpack to latest

### PR DESCRIPTION
**Pull request by ysz**
In order to add TypeScript support I've upgraded webpack and friends to the latests versions as it seems to be very out of date.

Unexpectedly validating mnemonic now fails here https://github.com/oasisprotocol/oasis-wallet-ext/blob/master/src/popup/pages/RestoreAccount/index.js#L48

At first I though it has to do with newer version of bip39 (it has not). In order to test it I created repro here https://github.com/ysz/bip39-3.0.4-test

I blame Buffer not being properly webpack'ed or something like this for bip39 (I'm investigating the issue)

Upgrade like this and run `yarn install` then:

```
$ npx npm-check-updates -u
npx: installed 314 in 13.537s
Using yarn
Upgrading /Users/ysz/oasis/code/oasis-wallet-ext/package.json
[====================] 55/55 100%

 @babel/plugin-proposal-class-properties   7.12.1  →   7.14.5     
 @babel/plugin-transform-react-jsx        7.12.12  →   7.14.9     
 @babel/plugin-transform-runtime           7.12.1  →   7.15.0     
 @babel/preset-env                         7.12.7  →   7.15.6     
 @babel/preset-react                       7.12.7  →   7.14.5     
 @babel/runtime                            7.12.5  →   7.15.4     
 babel-loader                               8.1.0  →    8.2.2     
 copy-webpack-plugin                        6.3.2  →    9.0.1     
 cspell                                    ^5.9.0  →   ^5.9.1     
 jest                                     ^26.6.3  →  ^27.2.0     
 jest-chrome                                0.7.1  →    0.7.2     
 jsdom                                     ^8.1.0  →  ^17.0.0     
 jsdom-global                               1.7.0  →    3.0.2     
 node-sass                                 4.14.1  →    6.0.1     
 react-scripts                              4.0.1  →    4.0.3     
 sinon                                    ^11.1.1  →  ^11.1.2     
 webpack                                   4.44.2  →   5.52.1     
 webpack-cli                                4.2.0  →    4.8.0     
 @ledgerhq/hw-transport-webusb             5.46.0  →    6.6.0     
 @oasisprotocol/ledger                      0.1.0  →    0.2.0     
 @reduxjs/toolkit                           1.4.0  →    1.6.1     
 axios                                     0.21.2  →   0.21.4     
 base64-arraybuffer                         0.2.0  →    1.0.1     
 bip39                                      3.0.2  →    3.0.4     
 classnames                                 2.2.6  →    2.3.1     
 i18next                                   19.8.4  →   20.6.1     
 react                                     17.0.1  →   17.0.2     
 react-circular-progressbar                 2.0.3  →    2.0.4     
 react-dom                                 17.0.1  →   17.0.2     
 react-i18next                             11.8.2  →  11.12.0     
 react-idle-timer                           4.2.5  →    4.6.4     
 react-redux                                7.2.2  →    7.2.5     
 react-router-dom                           5.1.2  →    5.3.0     
 redux                                      4.0.5  →    4.1.1     

Run yarn install to install new versions.
```

### CODE REVIEW

**Comment by pro-wh** [comment]
thanks for looking into that mnemonic thing.

@lvshaoping007 do you have any info on why the codebase up to now has stayed on webpack 4?

**Comment by pro-wh** [comment]
https://github.com/oasisprotocol/oasis-sdk/blob/5659e1b76cc65fa900232723a234f845903676e8/client-sdk/ts-web/core/playground/webpack.config.js#L9

hm and we use the same ProvidePlugin for Buffer in our other repos and it works there :shrug:

**Comment by tjanez** [comment]
> I blame Buffer not being properly webpack'ed or something like this for bip39 (I'm investigating the issue)

@ysz, we you able to debug this?

File: webpack.config.js [hunk 1]
```diff
@@ -125,9 +131,15 @@ function getPlugins(isDev) {
       React: "react",
     }),
     new webpack.DefinePlugin({
-      'process.env': JSON.stringify(process.env.NODE_ENV),
+      'process.env': { 
+        ...JSON.stringify(process.env.NODE_ENV),
```
Comment (lukaw3d) [inline webpack.config.js:135]: This can't be correct - spreading a string to object

File: src/popup/pages/RestoreAccount/index.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ class RestoreAccount extends React.Component {
       Toast.info(getLanguage('seedLengthError'))
       return
     }
+    console.log(`mnemonic is "${mnemonic}"`)
     let mnemonicValid = bip39.validateMnemonic(mnemonic)
```
Comment (lukaw3d) [inline src/popup/pages/RestoreAccount/index.js:49]: Putting debugger here and going into validateMnemonic https://github.com/bitcoinjs/bip39/blob/200c6a9/src/index.js#L148-L153
```js
function validateMnemonic(mnemonic, wordlist) {
    try {
        mnemonicToEntropy(mnemonic, wordlist);
    }
    catch (e) {
        return false;
```

`e` in catch is
```
TypeError: Cannot read properties of undefined (reading 'call')
    at Hash.CipherBase (popup.js:39643)
    at new Hash (popup.js:42098)
    at createHash (popup.js:42118)
    at deriveChecksumBits (popup.js:37288)
    at mnemonicToEntropy (popup.js:37345)
    at Object.validateMnemonic (popup.js:37394)
    at popup.js:26354
    at HTMLUnknownElement.callCallback (popup.js:62273)
    at Object.invokeGuardedCallbackDev (popup.js:62322)
    at invokeGuardedCallback (popup.js:62384)
```

that line comes from
https://github.com/crypto-browserify/cipher-base/blob/bc18dc84/index.js#L7
```js
var Transform = require('stream').Transform
..
function CipherBase (hashMode) {
  Transform.call(this)
```

It seems `stream` is not polyfilled correctly. It only has `require('stream').Stream`

Comment (lukaw3d) [reply src/popup/pages/RestoreAccount/index.js:49]: Compiled code before and after
```diff
- var Transform = __webpack_require__(/*! stream */ "./node_modules/stream-browserify/index.js").Transform
+ var Transform = __webpack_require__(/*! stream */ "./node_modules/stream/index.js").Transform
```

Comment (buberdds) [reply src/popup/pages/RestoreAccount/index.js]: Hey, seems like Oasis is leaving webpack world, but you could try sth like this https://github.com/ysz/oasis-wallet-ext/compare/upgrade-webpack...buberdds:upgrade-webpack 

There is also a new PR in `cipher-base` https://github.com/crypto-browserify/cipher-base/pull/16 which targets that issue.

**Pull request closed without merging**